Sources/Plasma/NucleusLib/pnAsyncCoreExe/pnAceSocket.cpp

```diff
@@ -125,7 +125,7 @@ struct WriteOperation
 
     asio::const_buffer AsBuffer() const
     {
-        return asio::buffer(fNotify.buffer, fNotify.bytes);
+        return asio::buffer(fNotify.buffer + fNotify.bytesCommitted, fNotify.bytes - fNotify.bytesCommitted);

```

Comment:
One issue with reusing buffers is it's not clear what the start and endpoints of the new data is, so it can cause the data length sent to the server to be wrong. bytes is the total size of the buffer, but that may include the length of data already sent.

It seems like Windows had a Win32 structure called OVERLAPPED that was part of socket I/O to deal with this problem and track which parts of the buffer were already committed. Don't need anything fancy, just need a `bytesCommitted` field to track how much of `bytes` has already been submitted.

`bytes` is still important because it's compared against `bytesProcessed`